openbb_platform/providers/intrinio/openbb_intrinio/models/market_indices.py

```diff
@@ -1,91 +1,91 @@
-
-
-class IntrinioMarketIndicesFetcher(
-    Fetcher[
-        IntrinioMarketIndicesQueryParams,
-        List[IntrinioMarketIndicesData],
-    ]
-):
-    """Transform the query, extract and transform the data from the Intrinio endpoints."""
-
-    @staticmethod
-    def transform_query(params: Dict[str, Any]) -> IntrinioMarketIndicesQueryParams:
-        """Transform the query params."""
-        transformed_params = params
-
-        now = datetime.now().date()
-        if params.get("start_date") is None:
-            transformed_params["start_date"] = now - relativedelta(years=1)
-
-        if params.get("end_date") is None:
-            transformed_params["end_date"] = now
-
-        return IntrinioMarketIndicesQueryParams(**transformed_params)
-
-    @staticmethod
-    def extract_data(
-        query: IntrinioMarketIndicesQueryParams,
-        credentials: Optional[Dict[str, str]],
-        **kwargs: Any,
-    ) -> List[Dict]:
-        """Return the raw data from the Intrinio endpoint."""
-        api_key = credentials.get("intrinio_api_key") if credentials else ""
-
-        base_url = "https://api-v2.intrinio.com/indices/stock_market"
-        query_str = get_querystring(query.model_dump(by_alias=True), ["symbol", "tag"])
-        url = f"{base_url}/{query.symbol}/historical_data/{query.tag}?{query_str}&api_key={api_key}"
-
-        return get_data_many(url, "historical_data", **kwargs)
-
-    @staticmethod
-    def transform_data(
-        query: IntrinioMarketIndicesQueryParams,  # pylint: disable=unused-argument
-        data: List[Dict],
-        **kwargs: Any,
-    ) -> List[IntrinioMarketIndicesData]:
-        """Return the transformed data."""
-        return [IntrinioMarketIndicesData.model_validate(d) for d in data]
+"""Intrinio Market Indices Model."""
+
+from datetime import datetime
+from typing import Any, Dict, List, Literal, Optional
+
+from dateutil.relativedelta import relativedelta
+from openbb_core.provider.abstract.fetcher import Fetcher
+from openbb_core.provider.standard_models.market_indices import (
+    MarketIndicesData,
+    MarketIndicesQueryParams,
+)
+from openbb_core.provider.utils.descriptions import QUERY_DESCRIPTIONS
+from openbb_core.provider.utils.helpers import get_querystring
+from openbb_intrinio.utils.helpers import get_data_many
+from pydantic import Field, NonNegativeInt
+
+
+class IntrinioMarketIndicesQueryParams(MarketIndicesQueryParams):
+    """Intrinio Market Indices Query.
+
+    Source: https://docs.intrinio.com/documentation/web_api/get_stock_market_index_historical_data_v2
+    """
+
+    tag: Optional[str] = Field(default="level", description="Index tag.")
+    type: Optional[str] = Field(
+        default=None,
+        description="Index type.",
+    )
+    sort: Literal["asc", "desc"] = Field(
+        default="desc",
+        description="Sort order.",
+        alias="sort_order",
+    )
+    limit: NonNegativeInt = Field(
+        default=100,
```

Comment:
I think while we are here, making this default to 10,000 will allow start/end dates to work better.

Reply:
It took ~4 min when the limit was 4000, 10000 would be ~10 min or so. 

Would need to add the async code in place of the `ThreadPoolExecutor` to improve time complexity. #5788 is taking care of this issue so I'll ask @tehcoderer to make the limit 10000 on his PR after this is merged to develop.

Reply:
The `start_date` and `end_date` should be taken into account if they are provided, regardless of the `limit`. This means a change in the code logic which is out of the scope of this PR.

Reply:
> It took ~4 min when the limit was 4000, 10000 would be ~10 min or so.

No, it allows the entire history to be returned by a single query.

![Screenshot 2023-11-27 at 1 17 50 PM](https://github.com/OpenBB-finance/OpenBBTerminal/assets/85772166/3ea9afc9-dba0-45ed-ae4c-c4ae6d67f4dd)

Reply:
I mistook this for the institutional ownership endpoint where it takes much more time to fetch the data.

But I will reiterate that the date logic should be implemented in the code which is more robust. The limit value by default is fine for now.

Reply:
As per the recent team's discussion, the API params are to be included in the provider model, whether its used or not.

`limit` param is here to stay and the better user experience would be to implement the logic for the dates, as mentioned.

Reply:
I'm simply asking to increase the limit to an arbitrarily high number because that allows results to return without the need for pagination.  The "logic" for dates already exists, increasing the `page_size` is all that needs to happen.

Reply:
On further testing, I find that 1000 is the maximum number of entries returned. I will set it as 1000 for the time being.

Reply:
How are you arriving at 1000 being the max?  I quite easily get 8000, no pagination required.

Here's the URL: 

"https://api-v2.intrinio.com/indices/stock_market/$NDX/historical_data/level?start_date=1990-11-27&end_date=2023-11-27&sort_order=desc&page_size=100000&api_key="

![Screenshot 2023-11-27 at 4 20 54 PM](https://github.com/OpenBB-finance/OpenBBTerminal/assets/85772166/58b80814-0cd6-4a40-a57c-7dc6db64be49)